### ISSUE TITLE
UI examples

### ISSUE THREAD

**Issue by littlehaker** [replit/kaboom#24]
Hi, Kaboom looks great, very simple API.

I'm new to game develop, it seems to be quite different from web develop, especially UI stuff.

Are there more examples about UI? Like Modal, Input, Dropdown etc.

**Comment by slmjkdbtl** [comment]
You can build UI using visual primitives like `rect()`, `sprite()`, there's a simple [button example](https://kaboomjs.com/examples#button), but it's a bit clunky, I'll think about how to make UI easier

**Comment by littlehaker** [comment]
Thanks, I've seen the button example, each button is composed by two objects. 

When I want to hide a button, I may need to manually operate these two objects. Perhaps a tag will be helpful but still a bit verbose. 

It goes worse when handling complex UI components like Modal or Overlay. 

I expect your idea for how to make it easier so much.

**Comment by soypunk** [comment]
Agreed that some guidance on what's best handled in kaboom and best left to some other web-tech would be helpful. For example if I have a game where someone needs to select "game pieces" from a list of several choices and then place it onto the "map layer" how much should be I trying to shoehorn this UI interaction and layout into kaboom -or- should I just make my own "drawer" (or other UI element) and make sure the objects in that drawer are recognized by kaboom in some form. (And when I have reached the point that kaboom is not really the right game library for the game I have in mind? That's not a faulting of kaboom in my mind, every library has its edges.)

**Comment by triptych** [comment]
> Agreed that some guidance on what's best handled in kaboom and best left to some other web-tech would be helpful. For example if I have a game where someone needs to select "game pieces" from a list of several choices and then place it onto the "map layer" how much should be I trying to shoehorn this UI interaction and layout into kaboom -or- should I just make my own "drawer" (or other UI element) and make sure the objects in that drawer are recognized by kaboom in some form. (And when I have reached the point that kaboom is not really the right game library for the game I have in mind? That's not a faulting of kaboom in my mind, every library has its edges.)

This might fall  into the category of plugin perhaps. I believe once the number of examples grows we'll see strategies for things like you want - without a lot of extra coding on your part.
